src/rocketsonde/core/probe.py

```diff
@@ -93,13 +101,9 @@ def stop_monitor(self, timeout=3):
         if self._worker is None:
             raise ValueError("Cannot stop when not running")
 
-        self._worker.shutdown_event.set()
+        self._worker.pid_queue.put(SHUTDOWN)
```

Comment:
I see, a queue, no longer a single event. Maybe queues play nicely with queues.

Reply:
This doesn't have the problem the events to. I don't trust this API as much as I did when I started this project, but I think this should be safe.

Reply:
I'm smiling about the "don't trust as much". Sounds good.